src/containers/Accounts/AMM/AMMAccounts/index.tsx

```diff
@@ -0,0 +1,220 @@
+import React, { useContext, useEffect, useState } from 'react'
+import { useParams, useRouteMatch } from 'react-router'
+import { useTranslation } from 'react-i18next'
+import { connect } from 'react-redux'
+import AMMAccountHeader from 'containers/Accounts/AMM/AMMAccounts/AMMAccountHeader/AMMAccountHeader'
+import { AccountTransactionTable } from 'containers/Accounts/AccountTransactionTable/index'
+import NoMatch from 'containers/NoMatch'
+import 'containers/Accounts/styles.scss'
+import {
+  analytics,
+  ANALYTIC_TYPES,
+  formatAsset,
+  formatBalance,
+} from 'containers/shared/utils'
+import { Tabs } from 'containers/shared/components/Tabs'
+import { AccountAssetTab } from 'containers/Accounts/AccountAssetTab/AccountAssetTab'
+import { getAccountTransactions, getAMMInfo } from 'rippled/lib/rippled'
+import SocketContext from '../../../shared/SocketContext'
+import { ERROR_MESSAGES } from '../../Errors'
+import { getTokenPairData } from '../../../../apis/OnTheDex'
+
+export interface AmmDataType {
+  balance: { currency: string; amount: number; issuer: string }
+  balance2: { currency: string; amount: number; issuer: string }
```

Comment:
Pretty sure these are already types somewhere (`IssuedCurrencyAmount` in `types.ts` I believe)

Reply:
The types differ slightly, IssuedCurrency doesnt have an amount field.

Reply:
https://github.com/ripple/explorer/blob/staging/src/containers/shared/types.ts#L12-L18
```
export type ExplorerAmount =
  | string
  | {
      issuer?: string
      currency: string
      amount: number
    }
```

Reply:
I think case the string version isn't allowed.  That portion could be cleaned up though so you could the type with issuer not required and then another type leveraging `Required<Type>`